### PR TITLE
[Merged by Bors] - feat: Summable.countable_support

### PR DESCRIPTION
**Pull request by nomeata**
A step towards showing that `Pmf`s have countable support.

Thanks Eric Rodriguez and Kevin Buzzard for helping on zulip.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Topology/Instances/ENNReal.lean [hunk 1]
```diff
@@ -1117,6 +1117,28 @@ theorem summable_of_le {f g : β → ℝ≥0} (hgf : ∀ b, g b ≤ f b) : Summa
     hp.summable
 #align nnreal.summable_of_le NNReal.summable_of_le
 
+/-- Summable non-negative functions have countable support -/
+theorem _root_.Summable.countable_support (f : α → ℝ≥0) (h : Summable f) :
```
Comment (eric-wieser) [inline Mathlib/Topology/Instances/ENNReal.lean:1121]:
```suggestion
theorem _root_.Summable.countable_support_nnreal (f : α → ℝ≥0) (h : Summable f) :
```
then you could provide a short wrapper for `real` that takes the hypothesis that `f a` is nonnegative everywhere.

Comment (nomeata) [reply Mathlib/Topology/Instances/ENNReal.lean]: Sure thing!

**Comment by urkud** [comment]
I'll try to generalize it today. If I don't push a generalization in 5h, then ignore this comment.

File: Mathlib/Topology/Instances/ENNReal.lean [hunk 2]
```diff
@@ -1302,6 +1324,12 @@ theorem Summable.toNNReal {f : α → ℝ} (hf : Summable f) : Summable fun n =>
   simp only [le_abs_self, Real.coe_toNNReal', max_le_iff, abs_nonneg, and_self_iff]
 #align summable.to_nnreal Summable.toNNReal
 
+/-- Summable non-negative functions have countable support -/
+theorem _root_.Summable.countable_support {f : α → ℝ} (hf : ∀ x, 0 ≤ f x) (h : Summable f) :
+  f.support.Countable := by
```
Comment (eric-wieser) [inline Mathlib/Topology/Instances/ENNReal.lean:1329]: Mathlib4 style is
```suggestion
    f.support.Countable := by
```

File: Mathlib/Topology/Instances/ENNReal.lean [hunk 3]
```diff
@@ -1117,6 +1117,28 @@ theorem summable_of_le {f g : β → ℝ≥0} (hgf : ∀ b, g b ≤ f b) : Summa
     hp.summable
 #align nnreal.summable_of_le NNReal.summable_of_le
 
+/-- Summable non-negative functions have countable support -/
+theorem _root_.Summable.countable_support_nnreal (f : α → ℝ≥0) (h : Summable f) :
+  f.support.Countable := by
```
Comment (eric-wieser) [inline Mathlib/Topology/Instances/ENNReal.lean:1122]:
```suggestion
    f.support.Countable := by
```

**Comment by ADedecker** [comment]
Thanks!

bors merge

**Comment by nomeata** [comment]
Thanks for the generalizations and the quick merge! Did you drop the Real variant intentionally? (I don't need it right now but Eric suggested it)

**Comment by ADedecker** [comment]
You can apply the general `Summable.countable_support` in the real case (but not the `NNReal`/`ENNReal` cases, which is why they have specialized lemmas). That said I should have waited for your approval before merging, sorry for that.

**Comment by nomeata** [comment]
Ah, of course, now I see it. And no worries!

**Pull request closed without merging**